python/invoke_release/tasks.py

```diff
@@ -1045,6 +1049,67 @@ def _post_rollback(current_version, rollback_to_version):
         plugin.post_rollback(ROOT_DIRECTORY, current_version, rollback_to_version)
 
 
+def _get_version_to_bump(changelog_message):
+    major_commit_present = None
+    minor_commit_present = None
+    patch_commit_present = None
+
+    for line in changelog_message:
+        if line.startswith('- ' + MAJOR_TAG):
+            major_commit_present = True
+        elif line.startswith('- ' + MINOR_TAG):
+            minor_commit_present = True
+        elif line.startswith('- ' + PATCH_TAG):
+            patch_commit_present = True
+        else:
+            # If a line in the changelog message doesn't start with a version
+            # tag, suggest nothing.
+            return None
+
+    version = PATCH_TAG if patch_commit_present else None
+    version = MINOR_TAG if minor_commit_present else version
+    version = MAJOR_TAG if major_commit_present else version
+
+    return version
+
+
+def _bump_major_version(current_version):
+    return (current_version[0] + 1, 0, 0)
+
+
+def _bump_minor_version(current_version):
+    return (current_version[0], current_version[1] + 1, 0)
+
+
+def _bump_patch_version(current_version):
+    return (current_version[0], current_version[1], current_version[2] + 1)
+
+
+def _suggest_version(current_version, version_to_bump):
+    current_version = tuple(
+        map(
+            int,
+            current_version.split('-')[0].split('+')[0].split('.')[:3]
+        )
+    )
+    suggested_version = None
+
+    if version_to_bump == PATCH_TAG:
+        suggested_version = _bump_patch_version(current_version)
+    if version_to_bump == MINOR_TAG:
+        suggested_version = _bump_minor_version(current_version)
+    if version_to_bump == MAJOR_TAG:
+        if current_version[0] == 0:
+            # For MAJOR version zero, recommend to bump a MINOR version instead
+            # since going for version 1.x.x should be a conscious decision and
+            # suggestion wouldn't be necessary.
+            suggested_version = _bump_minor_version(current_version)
+        else:
+            suggested_version = _bump_major_version(current_version)
+
+    return '.'.join(map(str, suggested_version)) if suggested_version else None
```

Comment:
This and the four versions above could all be one, shorter function with replacement logic driven by Enum values if you followed the Enum suggestion above.

Reply:
I _think_ I've followed your suggestion. If that's not what you meant I'm gonna need more explaining 😛

Reply:
I finally didn't go the `enum` way since that would make us add a new hard requirement (`enum34`) or lose compatibility with Python 2.7